src/services/EthService/nodes/web3/node.ts

```diff
@@ -104,10 +112,16 @@ export async function setupWeb3Node() {
       (window as any).web3 = new (window as any).Web3(ethereum);
     }
     const web3Node = new Web3Node();
-    const permissions = await requestPermission(web3Node);
-    if (permissions) {
+    const detectedPermissions = await getPermissions(web3Node);
```

Comment:
With this how am I supposed to add new MetaMask accounts? If I have already added accounts it never prompts me and just goes back to the dashboard.